### ISSUE TITLE
Side Web Panel not Loading Icons Correctly.

### ISSUE THREAD

**Issue by Mibix** [Floorp-Projects/Floorp#1221]
### Pre-Submission Checklist

- [X] I have checked the existing/closed issues to ensure this issue has not already been reported.
- [X] I am using the latest supported version of the browser.
- [X] I have tried disabling all browser extensions/add-ons to rule out interference from these.
- [X] I have tried clearing the browser cache and cookies.
- [X] I have checked with a new profile and the issue still occurs.
- [X] I have provided detailed step-by-step instructions on how to reproduce the issue.
- [X] I have included relevant screenshots or console outputs.
- [X] I have checked this problem is not occurring in the latest version of Firefox ESR.
- [X] This issue is specific to this browser and does not occur in other browsers.

### Bug Summary

Some sites don't load their icons properly and show up either blank or as just a globe in the side web panel.  You can force them with custom CSS if you know the web address of the favicon.ico but its not easy to find.  This is a lot less troublesome on Vivaldi.  Homeassistant instances are one where it can't find it for example.

### Reproduction Steps

1. Add new web panel
2. Add a site where the favicon.ico isn't in the root of the site.
3. Watch the icon be missing

### Expected Behavior

Should load the same icon that is displayed when the site is loaded in a tab.

### Actual Behavior

Loads a default web icon or no icon at all

### Screenshots

![Screenshot 2024-06-12 033705](https://github.com/Floorp-Projects/Floorp/assets/7806070/03f588d5-9e2f-451e-9f3d-051ed0d2df88)

### Browser Version

11.14.0 (64-bit)

### Operating System

Win 11 Version 22H2 Build 22621.3737

**Comment by nyanrus** [comment]
https://www.reddit.com/r/Floorp/s/IpDy6Y3NOn
This is because Floorp uses external "icon provider" to get favicon.
the way to fix is to allow to get favicon from server directly with IP(incl.ipv6) pattern.

**Comment by Mibix** [comment]
> this issue is from https://www.reddit.com/r/Floorp/s/IpDy6Y3NOn This is because Floorp uses external "icon provider" to get favicon. the way to fix is to allow customizing to get favicon from server directly with IP(incl.ipv6) pattern.

Yeah I got the workaround working but what is the advantage of having an external icon provider?

**Comment by nyanrus** [comment]
It is useful to get icon without accessing to the server.
It do not need to get the html, no accessing to server, no log to the server.

Some website seems svg with html ling tag, not favicon.ico.
And the external icon providers provide unified API to get these favicon.
https://developers.google.com/search/docs/appearance/favicon-in-search

**Comment by deletedpenguin** [comment]
Ideally it parses the website for the icon, but an alternative would be to be able to set a custom icon for websites. This would also solve for those local links or those that want to customise the icons.

**Comment by lufixSch** [comment]
> Ideally it parses the website for the icon, but an alternative would be to be able to set a custom icon for websites. This would also solve for those local links or those that want to customise the icons.

I'd really love that feature as I have a lot of custom/local webpages in that sidebar which often have the same or no favicon at all.

**Comment by Arch-Storm** [comment]
I personally would love to see some sort of custom icon method as my telegram icon get's a white square on google icon provider, but under duckduckgo the web.whatsapp icon just disappears.
Google:
![image](https://github.com/user-attachments/assets/dd9e9917-b87b-406c-bdea-f13cd82c6a78) 
DDG:
![image](https://github.com/user-attachments/assets/bf7a4ef5-8810-44cf-a332-fb7eb2d11e17)

**Comment by deletedpenguin** [comment]
> I personally would love to see some sort of custom icon method as my telegram icon get's a white square on google icon provider, but under duckduckgo the web.whatsapp icon just disappears. Google: ![image](https://private-user-images.githubusercontent.com/42775469/362211859-dd9e9917-b87b-406c-bdea-f13cd82c6a78.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4MzcxMTMsIm5iZiI6MTcyNDgzNjgxMywicGF0aCI6Ii80Mjc3NTQ2OS8zNjIyMTE4NTktZGQ5ZTk5MTctYjg3Yi00MDZjLWJkZWEtZjEzY2Q4MmM2YTc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDA5MjAxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2NTFjZDkyYzcwYTc3MjU0OWNlMjM5YjFmMDYyOGRhM2E0MTVhNjcyZjQ2NzUyODMxNDgxZDAzODI3NDliNTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mYv6b3F2f13wmVvP1Eqe-IzxHdgR30V7eWyh1U_O8Rw) DDG: ![image](https://private-user-images.githubusercontent.com/42775469/362212504-bf7a4ef5-8810-44cf-a332-fb7eb2d11e17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4MzcxMTMsIm5iZiI6MTcyNDgzNjgxMywicGF0aCI6Ii80Mjc3NTQ2OS8zNjIyMTI1MDQtYmY3YTRlZjUtODgxMC00NGNmLWEzMzItZmI3ZWIyZDExZTE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDA5MjAxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxMGRhNzliZjI3ZTgwN2M0NTk0ZDVkZTI1ZDAwMWM4M2UyMzgxYjZiMzcwMWY5OGMyNTBiNGNhZDI1ZDUwZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.N-JqBfsMoyS6y2m7glTg15RUIhKfDkhtEZNL0AHMrms)


This seems to be a new problem, as it didn't used to do this.
